docs/fri.md

```diff
@@ -4,12 +4,12 @@ FRI is a protocol that establishes that a committed polynomial has a bounded deg
 
 Since the codewords represent low-degree polynomials, and since the codewords are hidden behind Merkle trees in any real-world deployment, it is arguably more natural to present FRI from the point of view of a polynomial commitment scheme, with some caveats. There is scientific merit in separating the type of codewords from the IOP, and those two from the Merkle tree that simulates the oracles. However, from an accessibility point of view, it is beneficial to consider them as three components of one basic primitive that relates to polynomial commitment schemes. For the remainder of this tutorial, we will use the term FRI in this sense.
 
-In a regular polynomial commitment scheme, a prover commits to a polynomial $f(X)$ that is later opens in a given point $z$ such that it cannot equivocate between two different values of $f(z)$. The scheme consists of three algorithms:
+In a regular polynomial commitment scheme, a prover commits to a polynomial $f(X)$ that is later opened in a given point $z$ such that it cannot equivocate between two different values of $f(z)$. The scheme consists of three algorithms:
  - $\mathsf{commit}$, which computes a binding commitment from the polynomial;
  - $\mathsf{open}$, which produces a proof that $f(z) = y$ for some $z$ and for the polynomial $f(X)$ that matches with the given commitment;
  - $\mathsf{verify}$, which verifies the proof produced by $\mathsf{open}$.
 
-The FRI scheme has a different interface but a later section shows how it can simulate the standard polynomial commitment scheme interface without much overhead. FRI is a protocol between a prover and a verifier, which establishes that a given codeword belongs to a polynomial of low degree -- low meaning at most $\rho$ times the length of the codeword. Without losing much generality[^1], the prover knows this codeword explicitly, whereas the verifier knows only its Merkle root and leafs of his choosing, assuming the successful validation of the authentication paths that establish the leafs' membership to the Merkle tree.
+The FRI scheme has a different interface but a later section shows how it can simulate the standard polynomial commitment scheme interface without much overhead. FRI is a protocol between a prover and a verifier, which establishes that a given codeword belongs to a polynomial of low degree -- low meaning at most $\rho$ times the length of the codeword. Without losing much generality[^1], the prover knows this codeword explicitly, whereas the verifier knows only its Merkle root and leaves of his choosing, assuming the successful validation of the authentication paths that establish the leaves' membership to the Merkle tree.
```

Comment:
```suggestion
The FRI scheme has a different interface but a later section shows how it can simulate the standard polynomial commitment scheme interface without much overhead. FRI is a protocol between a prover and a verifier, which establishes that a given codeword belongs to a polynomial of low degree -- low meaning at most $\rho$ times the length of the codeword. Without losing much generality[^1], the prover knows this codeword explicitly, whereas the verifier knows only its Merkle root and leafs of his choosing, assuming the successful validation of the authentication paths that establish the leafs' membership to the Merkle tree.
```

I disagree with sources claiming that "leafs" is an invalid plural form of "leaf" and choose instead to pretend that the only reputable sources (e.g., [1]) do list "leafs" as a valid plural form. I prefer "leafs" over "leaves" particularly in the context of Merkle trees because "leaves" is also a frequently used verb (e.g., "she leaves the door open") and throws me off every time. Yes, "leafs" is *also* a valid verb, but it is not one that is used often and is therefore less likely to confuse.

[1]: https://www.merriam-webster.com/dictionary/leaf